### PR TITLE
Add a response forwarding section to the service documentation 

### PR DESCRIPTION
**Pull request by jbdoumenjou**
### What does this PR do?

Add a Response Forwarding section to the service documentation with a File Provider example.

### Motivation

To have a better documentation.

### More

- [ ] ~Added/updated tests~
- [x] Added/updated documentation

### CODE REVIEW

File: docs/content/routing/services/index.md [hunk 1]
```diff
@@ -321,7 +321,38 @@ Below are the available options for the health check mechanism:
                 My-Custom-Header: foo
                 My-Header: bar
     ```
+    
+#### Response Forwarding
+
+Configure response forwarding to define how forwards serves back the response of requests to remote server.
```
Comment (mpl) [inline docs/content/routing/services/index.md]: This section is about configuring how Traefik forwards the response from the backend server to the client.

File: docs/content/routing/services/index.md [hunk 2]
```diff
@@ -321,7 +321,38 @@ Below are the available options for the health check mechanism:
...
+
+Below are the available options for the Response Forwarding mechanism:
+
+- `FlushInterval` specifies the flush interval to flush to the client while copying the response body.
```
Comment (mpl) [inline docs/content/routing/services/index.md]:
```suggestion
- `FlushInterval` specifies the interval in between flushes to the client while copying the response body.
```

File: docs/content/routing/services/index.md [hunk 3]
```diff
@@ -321,7 +321,38 @@ Below are the available options for the health check mechanism:
...
+  It is a duration in milliseconds, defaulting to 100.
+  A negative value means to flush immediately after each write to the client.
+  The FlushInterval is ignored when ReverseProxy recognizes a response as a streaming response; for such responses, writes are flushed to the client immediately.
```
Comment (mpl) [inline docs/content/routing/services/index.md]:
```suggestion
  The FlushInterval is ignored when ReverseProxy recognizes a response as a streaming response;
```

File: docs/content/routing/services/index.md [hunk 4]
```diff
@@ -321,7 +321,38 @@ Below are the available options for the health check mechanism:
...
+
```
Comment (mpl) [inline docs/content/routing/services/index.md]:
```suggestion
  for such responses, writes are flushed to the client immediately.
  
```

File: docs/content/routing/services/index.md [hunk 5]
```diff
@@ -321,7 +321,38 @@ Below are the available options for the health check mechanism:
...
+Configure response forwarding to define how forwards serves back the response of requests to remote server.
+
+Below are the available options for the Response Forwarding mechanism:
+
+- `FlushInterval` specifies the flush interval to flush to the client while copying the response body.
+  It is a duration in milliseconds, defaulting to 100.
+  A negative value means to flush immediately after each write to the client.
+  The FlushInterval is ignored when ReverseProxy recognizes a response as a streaming response; for such responses, writes are flushed to the client immediately.
+  
+??? example "Use a custom FlushInterval -- Using the [File Provider](../../providers/file.md)"
```
Comment (mpl) [inline docs/content/routing/services/index.md]:
```suggestion
??? example "Using a custom FlushInterval -- Using the [File Provider](../../providers/file.md)"
```

File: docs/content/routing/providers/docker.md [hunk 6]
```diff
@@ -332,9 +332,8 @@ you'd add the label `traefik.http.services.<name-of-your-choice>.loadbalancer.pa
     ```
 
 ??? info "`traefik.http.services.<service_name>.loadbalancer.responseforwarding.flushinterval`"
-    <!-- TODO doc responseforwarding in services page -->
     
-    FlushInterval specifies the flush interval to flush to the client while copying the response body.
+    See [response fowarding](../services/index.md#response-forwarding) for more information.
```
Comment (dduportal) [inline docs/content/routing/providers/docker.md]:
```suggestion
    See [response forwarding](../services/index.md#response-forwarding) for more information.
```

File: docs/content/routing/providers/marathon.md [hunk 7]
```diff
@@ -252,10 +252,9 @@ For example, to change the passHostHeader behavior, you'd add the label `"traefi
     ```
 
 ??? info "`traefik.http.services.<service_name>.loadbalancer.responseforwarding.flushinterval`"
-    <!-- TODO doc responseforwarding in services page -->
-    
-    FlushInterval specifies the flush interval to flush to the client while copying the response body.
     
+    See [response fowarding](../services/index.md#response-forwarding) for more information.
```
Comment (dduportal) [inline docs/content/routing/providers/marathon.md]:
```suggestion
    See [response forwarding](../services/index.md#response-forwarding) for more information.
```

File: docs/content/routing/providers/rancher.md [hunk 8]
```diff
@@ -258,10 +258,9 @@ you'd add the label `traefik.http.services.{name-of-your-choice}.loadbalancer.pa
     ```
 
 ??? info "`traefik.http.services.<service_name>.loadbalancer.responseforwarding.flushinterval`"
-    <!-- TODO doc responseforwarding in services page -->
-    
-    FlushInterval specifies the flush interval to flush to the client while copying the response body.
     
+    See [response fowarding](../services/index.md#response-forwarding) for more information.
```
Comment (dduportal) [inline docs/content/routing/providers/rancher.md]:
```suggestion
    See [response forwarding](../services/index.md#response-forwarding) for more information.
```

**Pull request merged**